toolchains/linux.cmake

```diff
@@ -34,6 +34,9 @@ set(CORE_COMPILE_DEFS PLATFORM_LINUX)
 # load core library
 add_subdirectory(${PROJECT_SOURCE_DIR}/core)
 
+check_mapzen_api_key()
+add_definitions( -DMAPZEN_API_KEY="$ENV{MAPZEN_API_KEY}" )
```

Comment:
I would suggest putting this within the `if(APPLICATION)` block below, since the API key isn't actually required unless you're building the demo (rather than just the library).

It would also be nice if `check_mapzen_api_key()` obtained the key for us (with an in-out variable, see https://cmake.org/pipermail/cmake/2008-November/025423.html)